crates/wasm-compose/README.md

```diff
@@ -0,0 +1,187 @@
+    <a href="https://docs.rs/wasm-compose/"><img src="https://img.shields.io/static/v1?label=docs&message=wasm-compose&color=blue&style=flat-square" alt="docs.rs docs" /></a>
+  </p>
+</div>
+
+## Overview
+
+`wasm-compose` is a tool for composing [WebAssembly components](https://github.com/webassembly/component-model)
+from other WebAssembly components.
+
+## Implementation status
+
+A quick note on the implementation status of the component model
+proposal:
+
+__At this time of this writing, no WebAssembly runtimes have fully
+implemented the component model proposal.__
+
+__[Wasmtime](https://github.com/bytecodealliance/wasmtime)
+has implementation efforts underway to support it, but it's still a
+_work-in-progress_.__
+
+## Example
+
+See the [example](example/README.md) directory for a complete example
+of composing WebAssembly components together.
+
+## Usage
+
+With a configuration file present, run `wasm-compose` without any options to
+compose a component:
+
+```sh
+cargo run -p wasm-compose
+```
+
+## Configuration
+
+By default, `wasm-compose` looks for a configuration file named
+`wasm-compose.toml` or `wasm-compose.yml` in the current directory.
+
+The composition configuration currently has three main sections:
+
+* `imports` - specifies the component imports for the composed component.
+* `instantiations` - specifies the instantiations of an imported component.
+* `exports` - specifies the exports of the composed component.
+
+### Imports section
+
+Each import is given a name to refer to that import from the `instantiations`
+section.
+
+In this example, an import named `a` is defined for a component found at `./a.wasm`:
+
+```toml
+[imports.a]
+path = "a.wasm"
+```
+By default, the composed component will _import_ its component dependencies
+rather than embedding them in the composed component.
+
+To change this, set the `embed` field to `true`:
+
+```toml
+[imports.a]
+path = "a.wasm"
+embed = true
+```
+
+Additionally, the `--embed` option can be given to `wasm-compose` to embed
+_all_ imports into the composed component.
+
+It is currently necessary to embed _all_ imports to run
+the composed component in [Wasmtime](https://github.com/bytecodealliance/wasmtime).
+
+### Instantiations section
+
+The `instantiations` section defines how the imported
+components are instantiated in the composed component.
+
+By default, an instantiation looks for an import of the same name. To change
+what import is instantiated, use the `import` field:
+
+```toml
+[instantiations.foo]
+import = "bar"
+```
+
+Here the `foo` instantiation uses the `bar` import rather than the expected
+default of the `foo` import.
+
+To specify a dependency between instantiations, use the `dependencies` field:
+
+```toml
+[instantiations.a]
+[instantiations.b]
+dependencies = ["a"]
+```
+
+In the above example, import `a` is _default_ instantiated because it has no
```

Comment:
I've changed this to read as "implicitly instantiated" and called out that it can only occur when the component has no imports.

Let me know if I need to make the new configuration file format explainer clearer.